src/screens/Accounts/Accounts.js

```diff
@@ -295,39 +315,132 @@ class AccountsScreen extends React.Component<Props, State> {
-      : accounts.filter(({ type }) => type !== ACCOUNT_TYPES.SMART_WALLET);
-
-    const visibleAccountsWithBalance = visibleAccounts.map((acc) => {
-      const accountBalances = balances[acc.id] || {};
-      const balance = calculatePortfolioBalance(assets, rates, accountBalances);
-      return { ...acc, balance };
+
+    const visibleAccounts = this.visibleAccounts(accounts, smartWalletFeatureEnabled);
+    const hasAccount = userHasSmartWallet(accounts);
+    const showSmartWalletInitButton = !hasAccount && smartWalletFeatureEnabled;
+    const fiatCurrency = baseFiatCurrency || defaultFiatCurrency;
+
+    const isEthereumActive = !!activeNetwork && (
+      activeNetwork.id === BLOCKCHAIN_NETWORK_TYPES.ETHEREUM
+    );
+
+    const wallets = visibleAccounts.map((account: Account): ListItem => {
+      const { id, isActive, type } = account;
+      const accountBalances = balances[id];
+      const isActiveWallet = !!isActive && isEthereumActive;
+      const isSmartWallet = type === ACCOUNT_TYPES.SMART_WALLET;
+      let walletBalance = 0;
+
+      if (accountBalances) {
+        // TODO: improve balance calculation?
+        const balance = calculatePortfolioBalance(assets, rates, accountBalances);
+        walletBalance = balance && balance[fiatCurrency];
+      }
+
+      const accountItem = {
+        id: `ACCOUNT_${id}`,
+        account,
+        type: 'ACCOUNT',
+        interaction: {
+          walletBalance,
+          onMainPress: () => this.switchWallet(account),
+          onSettingsPress: () => this.accountSettings(account),
+          isActiveWallet,
+          isSmartWallet,
+        },
+      };
+
+      return accountItem;
     });
 
-    const walletsToShow = showSmartWalletInitButton
-      ? [
-        ...visibleAccountsWithBalance,
-        { type: 'SMART_WALLET_INIT', action: () => navigation.navigate(SMART_WALLET_INTRO) },
-      ]
-      : visibleAccountsWithBalance;
+    if (showSmartWalletInitButton) {
+      wallets.push({
+        type: 'NEW_SMART_WALLET',
+        interaction: {
+          isSmartWallet: true,
+          walletBalance: 0,
+          isActiveWallet: false,
+          onMainPress: () => { navigation.navigate(SMART_WALLET_INTRO); },
+        },
+      });
+    }
+
+    return wallets;
+  }
+
+  networks(): ListItem[] {
+    const networks: ListItem[] = [];
+
+    const {
+      blockchainNetworks,
+      availableStake,
+      isTankInitialised,
+      smartWalletFeatureEnabled,
+    } = this.props;
+
+    const ppnNetwork = blockchainNetworks.find((network) => network.id === BLOCKCHAIN_NETWORK_TYPES.PILLAR_NETWORK);
+    const availableStakeFormattedAmount = formatMoney(availableStake, 4);
+
+    if (smartWalletFeatureEnabled && ppnNetwork) {
+      networks.push({
+        id: `NETWORK_${ppnNetwork.id}`,
+        type: 'NETWORK',
+        isInitialised: isTankInitialised,
+        network: ppnNetwork,
+        formattedStakeAmount: `${availableStakeFormattedAmount} ${PPN_TOKEN}`,
```

Comment:
Just a note: this could be renamed into balance, so this property could be reusable for other networks :)